src/Razor/Microsoft.AspNetCore.Razor.Tools/src/GenerateCommand.cs

```diff
@@ -227,6 +228,7 @@ protected override bool ValidateArguments()
             });
 
             var results = GenerateCode(engine, sourceItems);
+            var isGeneratingDeclaration = GenerateDeclaration.HasValue();
```

Comment:
Is there a particular reason the `GenerateDeclaration.HasValue` is cached here instead of called in the if-statement?

Reply:
It does a linq `Any()` call. It seems unnecessary to perform n-times (once for each code file we're code-generating).